lib/propmangle.js

```diff
@@ -159,7 +162,7 @@ function mangle_properties(ast, options) {
         if (options.only_cache) {
             return cache.props.has(name);
         }
-        if (/^[0-9.]+$/.test(name)) return false;
+        if (/^-?[0-9]+(\.[0-9]+)?(e[+-][0-9]+)?$/.test(name)) return false;
```

Comment:
The `e` should be `[Ee]` shouldn't it? 

```
$ node -p '({ 1E5: 1 })'
{ '100000': 1 }
```

Did you write this number regexp off the cuff or did you grab it from an official ECMAScript grammar and/or parser? I ask because whenever I tried to reformulate this number regexp I invariably get some detail wrong.